File: panel_sistema.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

// Estructura fija para compatibilidad con archivos binarios
typedef struct {
    int id;
    char nombre[40];
    int activo;
    float carga;
} Modulo;

void guardar_estado(const char *ruta, Modulo *lista, int total) {
    FILE *archivo = fopen(ruta, "wb");
    if (archivo != NULL) {
        fwrite(lista, sizeof(Modulo), total, archivo);
        fclose(archivo);
    }
}

void cargar_estado(const char *ruta, Modulo *lista, int total) {
    FILE *archivo = fopen(ruta, "rb");
    if (archivo != NULL) {
        fread(lista, sizeof(Modulo), total, archivo);
        fclose(archivo);
    }
}

int main() {
    const char *archivo_datos = "datos_c.dat";
    system("clear");
    
    // Lista de módulos base
    Modulo lista[] = {
        {1, "Compilador GCC Clang", 0, 12.5f},
        {2, "Entorno Mono C#", 0, 45.2f},
        {3, "Linter y Formateador Ruff", 0, 5.0f},
        {4, "Gestor de Paquetes PIP", 0, 18.3f},
        {5, "Analizador Estático Pylint", 0, 22.1f}
    };
    
    int total = sizeof(lista) / sizeof(lista[0]);
    cargar_estado(archivo_datos, lista, total);

    printf("=======================================================\n");
    printf("  GESTOR DE ENTORNO EN C — TERMUX AArch64\n");
    printf("=======================================================\n");
    
    for(int i = 0; i < total; i++) {
        printf("  [%d] %-30s %s  (%.1f%%)\n",
            lista[i].id, lista[i].nombre,
            lista[i].activo ? "✅ ACTIVO " : "⏸️ INACTIVO",
            lista[i].carga);
    }
    printf("=======================================================\n");

    int op;
    printf("\nSelecciona N° para alternar estado (0=Salir): ");
    if(scanf("%d", &op) == 1 && op >= 1 && op <= total) {
        lista[op-1].activo = !lista[op-1].activo;
        guardar_estado(archivo_datos, lista, total);
        printf("\n✅ Cambios guardados en '%s'.\n", archivo_datos);
    }
    
    return 0;
}

File: panel_sistema.cs
using System;
using System.IO;
using System.Collections.Generic;

class Modulo {
    public string Nombre { get; set; }
    public bool Activo { get; set; }
    public float Carga { get; set; }

    public Modulo(string n, float c) {
        Nombre = n; Carga = c; Activo = false;
    }
}

class Program {
    static string rutaArchivo = "datos_cs.txt";

    static void Main() {
        Console.Clear();
        var modulos = new List<Modulo> {
            new Modulo("Compilador GCC Clang", 12.5f),
            new Modulo("Entorno Mono C#", 45.2f),
            new Modulo("Linter y Formateador Ruff", 5.0f),
            new Modulo("Gestor de Paquetes PIP", 18.3f),
            new Modulo("Analizador Estático Pylint", 22.1f)
        };

        if (File.Exists(rutaArchivo)) {
            string[] lineas = File.ReadAllLines(rutaArchivo);
            for (int i = 0; i < modulos.Count && i < lineas.Length; i++) {
                if (bool.TryParse(lineas[i], out bool estado)) {
                    modulos[i].Activo = estado;
                }
            }
        }

        Console.WriteLine("=======================================================");
        Console.WriteLine("  GESTOR DE ENTORNO EN C# — MONO");
        Console.WriteLine("=======================================================");
        
        for (int i = 0; i < modulos.Count; i++) {
            Console.WriteLine($"  [{i + 1}] {modulos[i].Nombre,-30} {(modulos[i].Activo ? "✅ ACTIVO " : "⏸️ INACTIVO")}  Carga: {modulos[i].Carga:F1}%");
        }
        Console.WriteLine("=======================================================");

        Console.Write("\nSelecciona N° para cambiar estado (0=Salir): ");
        string entrada = Console.ReadLine();
        if (int.TryParse(entrada, out int op) && op >= 1 && op <= modulos.Count) {
            modulos[op - 1].Activo = !modulos[op - 1].Activa;

            List<string> estados = new List<string>();
            foreach (var mod in modulos) {
                estados.Add(mod.Activo.ToString());
            }
            File.WriteAllLines(rutaArchivo, estados);
            Console.WriteLine($"\n✅ Cambios guardados en '{rutaArchivo}'.");
        }
    }
}
